### DISCUSSION TITLE
GitHub code search Indexed sort "Last indexed" time API availability

### DISCUSSION THREAD

**Discussion by charlieley** [community/community#45132] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

While performing a Code search via UI, sorted by Indexed, such as in here - 
[https://github.com/search?o=desc&p=1&q=clam&s=indexed&type=Code](url)
we get a "last indexed" time, as shown here
![image](https://user-images.githubusercontent.com/101309988/214012957-dea9450e-b53b-4ec5-87a8-8692be62101a.png)

Is it possible to retrieve this "Last indexed" time via API in any iteration of GitHub code search API? If not, is this information available in any other way apart from the Web UI search?

**Comment by jge162** [comment]
GitHub Search API, allows you to search across all of GitHub. Search API will return a list of results that match the query, including the repository name, path, and the last update time of the file. (which is what you are looking for I believe).

Search API documentation can be found here: https://docs.github.com link doesn’t work. Try Google search for search api. 

while using Search API to search for the files you are interested in and then check the 

```python
updated_at
```
 field.

**Reply by charlieley** [reply]
Thanks for the reply. However, I am not sure that's the information that I need.
If I check the GitHub REST API documentation for code search - available here - https://docs.github.com/en/rest/search?apiVersion=2022-11-28#search-code
..it seems the "updated_at" field specified in the response schema only applies to the repository, not to the specific file which contains the search results.

Further, I am not certain the "last indexed" time as shown in the GitHub web search UI is the same as "updated_at" time available for a file / repository. I have a feeling, though I could be wrong, that the "last indexed" time can change, depending on when and why the indexer visited /re-visited it (information on specific conditions that lead to indexing / re-indexing isn't clear), while the "updated_at" time for a file / repo may not.

I'm specifically interested in the  "last indexed" time as can be shown in the web UI for code search.

**Reply by jge162** [reply]
Yeah that’s a tough one.

**Comment by look** [comment]
@charlieley you can add `sort:indexed` to your query to sort by indexed time. See [the documentation](https://docs.github.com/en/rest/search?apiVersion=2022-11-28#considerations-for-code-search).

**Reply by charlieley** [reply]
I do sort by `sort:indexed`, however it doesn't give me the specific "last indexed" time as seen in the web UI. Is there a way to get that time via the API?

**Reply by jge162** [reply]
@charlieley I looked and I do not see the option via the API to see "last indexed." its not on the documentation, Just says best results. https://docs.github.com/en/rest/search?apiVersion=2022-11-28#ranking-search-results
not the answer you were looking for but, closure. bests maybe ('Luke') can double check too.

**Reply by jge162** [reply]
<img width="520" alt="Screen Shot 2023-01-26 at 10 03 45 PM" src="https://user-images.githubusercontent.com/31228460/215020279-aa9d0452-0bd2-4142-96cb-4e70fdcb8a15.png">
 actually it does say "how" indexed

**Reply by charlieley** [reply]
@jge162 This is what I try, but there's no mention of a time or a time related field in the results.
[gh api --method=GET "search/code?q=clam&sort=indexed"](url)
Doesn't seem like there's a way to get that time via API...

**Reply by jge162** [reply]
> @jge162 This is what I try, but there's no mention of a time or a time related field in the results. [gh api --method=GET "search/code?q=clam&sort=indexed"](url) Doesn't seem like there's a way to get that time via API...

I think that is your answer there.

**Reply by charlieley** [reply]
I wonder if it'd be a huge effort to include that piece of information within the API. If the web UI search and gh API are working differently, then maybe a lot of work, but otherwise if they're all looking at the same piece of data, then it can be included in the API...?

**Comment by UmerYousuf** [comment]
Is there any way to increase pagination results and enable sorting?

**Reply by look** [reply]
[Not a this time](https://github.blog/changelog/2023-03-10-changes-to-the-code-search-api/).

We plan to release an API that allows exhaustive search in the future. Please let us know your use case, as that will help our planning for this.

**Reply by UmerYousuf** [reply]
Thank God you at least responded. Recently, indexed sorting and pagination have become the most important parts for me and other users. As a result, only five pages are shown, which are too old and can't display recently indexed codes. Please, please, please, try to update this ASAP. It has cost me a lot in my work. Looking forward to your positive response.

**Reply by look** [reply]
As I've mentioned in other threads, sorting by indexed has been removed because we rebuild the index too frequently for it to produce useful results -- it would be confusing. Increasing the pagination limit on the web site is on our backlog (the API supports 1000 results currently). 

I'm still interested to learn what your use case is that this is disrupting so we can consider it in future work.

**Reply by UmerYousuf** [reply]
Understood, so you meant an API supports 1000 pages of results. However, Is there any way to search code which recently indexed in any public repo? For example, I want to search "function Github" which is recently indexed in any public code. As I used to search code to understand the methodologies of different programmers through GitHub. As of now I search for any code it showed up as years old code which is completely used less for me and for others as the actual purpose of code search is to search recently indexed codes. Any tips or Future plans? And is there Github Enterprises still using the old GitHub search?

**Comment by look** [comment]
This question refers to the now removed code search system, so I'm going to close the discussion. If you would like to discuss sorting, please use this discussion: https://github.com/orgs/community/discussions/52932
